### PR TITLE
feat(frontend): Add animated three dots indicator for agent response

### PR DESCRIPTION
**Pull request by amanape**
Need suggestions: https://opendevin.slack.com/archives/C06TL7AJMK2/p1715848998465029

[Created with Copilot Workspace]
Related to #1787

Implements an animated three dots indicator in the chat UI to signal when the agent is generating a response.

- Adds a conditional rendering block within `ChatInterface.tsx` to display the animated three dots indicator using the `ChatMessage` component when the `curAgentState` is set to `AgentState.LOADING`.
- The indicator appears in the chat UI immediately after a user sends a message and is removed when the agent's response is ready to be displayed, enhancing user experience by providing visual feedback during response generation.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/OpenDevin/OpenDevin/issues/1787?shareId=e494f719-a835-4327-beee-530de0b240b5).

### CODE REVIEW

**Comment by amanape** [comment]
Updates the ChatInterface to render the ChatMessage component when the AgentState is RUNNING and adds custom spacing to the ChatMessage component.

- Changes the condition in `ChatInterface.tsx` to render `ChatMessage` when `AgentState` is `RUNNING` instead of `LOADING`.
- Adds a `className` prop to `ChatMessage` in `ChatInterface.tsx` to apply custom spacing, specifically a margin-top, ensuring consistent spacing with other messages.
- Modifies `ChatMessage.tsx` to accept an optional `className` prop for custom styling, allowing for more flexible styling adjustments in the future.


---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/OpenDevin/OpenDevin/pull/1821?shareId=07dabb40-1549-491b-ac63-8aa890321e02).

File: frontend/src/components/chat/ChatMessage.tsx [hunk 1]
```diff
@@ -5,14 +5,16 @@ import { code } from "../markdown/code";
...
 }
 
-function ChatMessage({ message }: MessageProps) {
+function ChatMessage({ message, className: customClassName }: MessageProps) {
   // const text = useTyping(message.content);
 
   const className = twMerge(
     "p-3 text-white max-w-[90%] overflow-y-auto rounded-lg",
     message.sender === "user" ? "bg-neutral-700 self-end" : "bg-neutral-500",
+    customClassName // Apply custom className if provided
```
Comment (yufansong) [inline frontend/src/components/chat/ChatMessage.tsx:17]:
```suggestion
    customClassName // Apply custom className if provided,
```

**Review by yufansong — APPROVED** [review]
LGTM, but plz fix the lint problem before merge.

File: frontend/src/components/chat/ChatMessage.tsx [hunk 2]
```diff
@@ -5,14 +5,16 @@ import { code } from "../markdown/code";
...
 }
 
-function ChatMessage({ message }: MessageProps) {
+function ChatMessage({ message, className: customClassName }: MessageProps) {
   // const text = useTyping(message.content);
 
   const className = twMerge(
     "p-3 text-white max-w-[90%] overflow-y-auto rounded-lg",
     message.sender === "user" ? "bg-neutral-700 self-end" : "bg-neutral-500",
+    customClassName // Apply custom className if provided,
```
Comment (yufansong) [inline frontend/src/components/chat/ChatMessage.tsx:17]:
```suggestion
    customClassName, // Apply custom className if provided
```

**Comment by amanape** [comment]
@yufansong Sorry I forgot to set this as a draft. Was using this issue to use the Copilot Workspace. I've updated the description to include the Slack message that briefly discusses possible alternatives to this issue. I don't think the 3-dots is one that will stick around

**Comment by Shimada666** [comment]
Looking forward to this feature! Hoping to enhance the user experience with the 3 dots indicator until we have a better option.

**Comment by amanape** [comment]
Stale

**Pull request closed without merging**